R/new.R

```diff
@@ -13,13 +13,22 @@ generator_funs$new <- function(...) {
     if (!identical(portable, inherit$portable))
       stop("Sub and superclass must both be portable or non-portable.")
 
+    # If `cloneable` property differs between sub and superclass
+    # - super will override sub if super doesn't allow cloning
+    # - sub will override super if super allows cloning
     if (!identical(cloneable, inherit$cloneable)) {
-      message(c(
-        "Sub and superclass have different cloneable properties. ",
-        "Subclass property will override superclass property."
-      ))
+      if (inherit$cloneable) {
+        inherit[["public_methods"]][["clone"]] <- NULL
```

Comment:
I think this will _modify_ the superclass, which is bad.

Reply:
Ah, good catch!

I have added a test for this, which is going to fail for now. Will revisit this tomorrow.

Reply:
This is now fixed.